test/e2e/router_status_test.go

```diff
@@ -0,0 +1,297 @@
+	ic.Spec.NamespaceSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"type": "foo",
+		},
+	}
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Create a new namespace for the route that we can immediately match with the IC's namespace selector.
+	nsFoo := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			Labels: map[string]string{
+				"type": "foo",
+			},
+		},
+	}
+	if err := kclient.Create(context.TODO(), nsFoo); err != nil {
+		t.Fatalf("failed to create namespace: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), nsFoo); err != nil {
+			t.Fatalf("failed to delete test namespace %v: %v", nsFoo.Name, err)
+		}
+	}()
+
+	// Create a new namespace for the route that we can NOT immediately match with the IC's namespace selector.
+	nsBar := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "bar",
+			Labels: map[string]string{
+				"type": "bar",
+			},
+		},
+	}
+	if err := kclient.Create(context.TODO(), nsBar); err != nil {
+		t.Fatalf("failed to create namespace: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), nsBar); err != nil {
+			t.Fatalf("failed to delete test namespace %v: %v", nsBar.Name, err)
+		}
+	}()
+
+	// Create a route to be immediately admitted by this ingress controller and then when the IC label selectors are
+	// updated, the status should clear.
+	routeFooLabelName := types.NamespacedName{Namespace: nsFoo.Name, Name: "route-foo-label"}
+	routeFooLabel := newRouteWithLabel(routeFooLabelName, "")
+	if err := kclient.Create(context.TODO(), routeFooLabel); err != nil {
+		t.Fatalf("failed to create route: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), routeFooLabel); err != nil {
+			t.Fatalf("failed to delete route %s: %v", routeFooLabelName, err)
+		}
+	}()
+
+	// Create a route that will NOT be immediately admitted by the ingress controller, but will be admitted AFTER
+	// the IC selectors are updated. The status SHOULD be successfully admitted.
+	routeBarLabelName := types.NamespacedName{Namespace: nsBar.Name, Name: "route-bar-label"}
+	routeBarLabel := newRouteWithLabel(routeBarLabelName, "bar")
+	if err := kclient.Create(context.TODO(), routeBarLabel); err != nil {
+		t.Fatalf("failed to create route: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), routeBarLabel); err != nil {
+			t.Fatalf("failed to delete route %s: %v", routeBarLabel, err)
+		}
+	}()
+
+	// Wait for routeFooLabel to be admitted upon creation.
+	admittedCondition := routev1.RouteIngressCondition{Type: routev1.RouteAdmitted, Status: corev1.ConditionTrue}
+	if err := waitForRouteIngressConditions(t, kclient, routeFooLabelName, ic.Name, admittedCondition); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
```

Comment:
Would it make sense also to check ` 	if err := waitForRouteStatusClear(t, kclient, routeBarLabelName, ic.Name)` here?